### DISCUSSION TITLE
OAuth flow

### DISCUSSION THREAD

**Discussion by pjsgsy** [danieldevine/bird-elephant#39] [Q&A]
Hi - Sorry to have to ask this, but I've been searching and reading for a few hours, and am stuck!  The library looks great. I am trying to post a simple 'Hello World' tweet to my Twitter account. I think I am struggling with the OAuth tokens required to do this.  I have the developer access and keys, etc.

It doesn't seem to be pointed out anywhere clearly, but, the library as is, with the developer access keys only mostly 'reads' tweets (maybe my assumption here is completely wrong).  To Post a tweet, per the 'simple' example here https://birdelephant.com/, you'd need to go through some sort of OAuth flow to get the token_identifier & token_secret that includes a callback to a publicly available script on my side?  Any pointers on how to do this anywhere?

[OK - I came back after a break and found the authenticate.php example....   Thanks!]

Thanks.

**Comment by danieldevine** [comment]
Just to be clear - Bird Elephant picks up at the end of the auth process - it will take whatever you give it in the credentials array and do its best to retrieve that data for you. (it'll also let you know if you have insufficient auth in the exceptions it returns), but it doesn't go about getting those credentials for you.    

This is intentional - oAuth is, I feel, best left to dedicated oAuth libraries.  This leaves Bird Elephant free to be as flexible as possible without tying the library's users into a predefined process.  

That said, I get that it can be difficult to master the auth part so I might consider bundling it in at some point, or writing some tutorials if that's something that people want.
